src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -330,10 +481,11 @@ ExecuteTaskAndStoreResults(QueryDesc *queryDesc, Task *task,
 		PlannedStmt *planStatement = queryDesc->plannedstmt;
 		MultiPlan *multiPlan = GetMultiPlan(planStatement);
 		Query *query = multiPlan->workerJob->jobQuery;
+		Oid relid = ((RangeTblEntry *) linitial(query->rtable))->relid;
 		StringInfo queryStringInfo = makeStringInfo();
 
 		ExecuteMasterEvaluableFunctions(query);
-		DeparseShardQuery(query, task, queryStringInfo);
+		deparse_shard_query(query, relid, task->anchorShardId, queryStringInfo);
```

Comment:
Hm, not that I think it matters much, but that function had intentionally been introduced by @lithp. Why did you roll back that change?